File: CleverOnion-backend/src/main/java/com/cleveronion/blog/domain/article/valueobject/ArticleStatus.java
package com.cleveronion.blog.domain.article.valueobject;

/**
 * 文章状态值对象
 * 表示文章的发布状态，是一个不可变的值对象
 * 
 * @author CleverOnion
 */
public enum ArticleStatus {
    
    /**
     * 草稿状态 - 文章尚未发布，仅作者可见
     */
    DRAFT("草稿"),
    
    /**
     * 已发布状态 - 文章已公开发布，所有用户可见
     */
    PUBLISHED("已发布"),
    
    /**
     * 已归档状态 - 文章已归档，不在主要列表中显示
     */
    ARCHIVED("已归档");
    
    private final String description;
    
    ArticleStatus(String description) {
        this.description = description;
    }
    
    /**
     * 获取状态描述
     * 
     * @return 状态描述
     */
    public String getDescription() {
        return description;
    }
    
    /**
     * 判断是否为已发布状态
     * 
     * @return 如果是已发布状态返回true，否则返回false
     */
    public boolean isPublished() {
        return this == PUBLISHED;
    }
    
    /**
     * 判断是否为草稿状态
     * 
     * @return 如果是草稿状态返回true，否则返回false
     */
    public boolean isDraft() {
        return this == DRAFT;
    }
    
    /**
     * 判断是否为归档状态
     * 
     * @return 如果是归档状态返回true，否则返回false
     */
    public boolean isArchived() {
        return this == ARCHIVED;
    }
}

File: CleverOnion-blog-k/src/main/kotlin/com/cleveronion/domain/article/valueobject/ArticleStatus.kt
package com.cleveronion.domain.article.valueobject

/**
 * 文章状态值对象
 * 
 * 表示文章的发布状态，包含状态转换规则
 */
enum class ArticleStatus {
    /**
     * 草稿状态 - 文章正在编写中，未发布
     */
    DRAFT,
    
    /**
     * 已发布状态 - 文章已公开发布
     */
    PUBLISHED,
    
    /**
     * 已归档状态 - 文章已归档，不再显示
     */
    ARCHIVED;
    
    /**
     * 检查是否可以转换到指定状态
     * 
     * 状态转换规则：
     * - DRAFT -> PUBLISHED, ARCHIVED
     * - PUBLISHED -> ARCHIVED
     * - ARCHIVED -> 无法转换到其他状态
     */
    fun canTransitionTo(newStatus: ArticleStatus): Boolean {
        return when (this) {
            DRAFT -> newStatus in setOf(PUBLISHED, ARCHIVED)
            PUBLISHED -> newStatus == ARCHIVED
            ARCHIVED -> false
        }
    }
    
    /**
     * 检查是否为已发布状态
     */
    fun isPublished(): Boolean = this == PUBLISHED
    
    /**
     * 检查是否为草稿状态
     */
    fun isDraft(): Boolean = this == DRAFT
    
    /**
     * 检查是否为已归档状态
     */
    fun isArchived(): Boolean = this == ARCHIVED
    
    /**
     * 检查是否为公开可见状态
     */
    fun isPubliclyVisible(): Boolean = this == PUBLISHED
    
    /**
     * 获取状态的中文描述
     */
    fun getDisplayName(): String {
        return when (this) {
            DRAFT -> "草稿"
            PUBLISHED -> "已发布"
            ARCHIVED -> "已归档"
        }
    }
    
    companion object {
        /**
         * 从字符串创建ArticleStatus
         */
        fun fromString(value: String): ArticleStatus {
            return try {
                valueOf(value.uppercase())
            } catch (e: IllegalArgumentException) {
                throw IllegalArgumentException("Invalid article status: $value")
            }
        }
        
        /**
         * 获取所有可公开访问的状态
         */
        fun getPublicStatuses(): Set<ArticleStatus> = setOf(PUBLISHED)
        
        /**
         * 获取所有可编辑的状态
         */
        fun getEditableStatuses(): Set<ArticleStatus> = setOf(DRAFT)
    }
}